tests/admin_views/test_autocomplete_view.py

```diff
@@ -0,0 +1,174 @@
+            AutocompleteJsonView.as_view(admin_site=site)(request)
+
+    def test_not_model_not_registered(self):
+        site.unregister(Question)
+        factory = RequestFactory()
+        request = factory.get(self.url, {'field_identifier': 'admin_views.Answer.question'})
+        request.user = self.superuser
+        with self.assertRaisesMessage(
+            Http404,
+            "<class 'admin_views.models.Question'> is not registered in the admin."
+        ):
+            AutocompleteJsonView.as_view(admin_site=site)(request)
+        site.register(Question, QuestionAdmin)
+
+    def test_success(self):
+        q = Question.objects.create(question='Is this a question?')
+        factory = RequestFactory()
+        request = factory.get(self.url, {'field_identifier': 'admin_views.Answer.question', 'term': 'is'})
+        request.user = self.superuser
+        response = AutocompleteJsonView.as_view(admin_site=site)(request)
+        self.assertEqual(response.status_code, 200)
+        data = json.loads(response.content.decode('utf-8'))
+        self.assertIn({'id': text_type(q.pk), 'text': q.question}, data['results'])
+
+    def test_permission_wrapper(self):
+        response = self.client.get(self.url, {'field_identifier': 'admin_views.Answer.question', 'term': ''})
+        self.assertEqual(response.status_code, 200)
+
+        self.client.logout()
+        response = self.client.get(self.url, {'field_identifier': 'admin_views.Answer.question', 'term': ''})
+        self.assertEqual(response.status_code, 302)
+
+    def test_has_change_permission_denied(self):
+        factory = RequestFactory()
+        request = factory.get(self.url, {'field_identifier': 'admin_views.Answer.question', 'term': 'is'})
+        self.user.is_staff = True
+        self.user.save()
+        request.user = self.user
+        with self.assertRaisesMessage(PermissionDenied, ""):
+            AutocompleteJsonView.as_view(admin_site=site)(request)
+
+    def test_has_change_permission_success_with_staff_user(self):
+        factory = RequestFactory()
+        request = factory.get(self.url, {'field_identifier': 'admin_views.Answer.question', 'term': 'is'})
+        p = Permission.objects.filter(codename='change_answer')
+        self.user.user_permissions.set(p)
+        self.user.is_staff = True
+        self.user.save()
+        self.user = get_user_model().objects.get(pk=self.user.pk)
+        self.assertTrue(self.user.has_perm('admin_views.change_answer'))
+        request.user = self.user
+
+        response = AutocompleteJsonView.as_view(admin_site=site)(request)
+        self.assertEqual(response.status_code, 200)
+
+
+@override_settings(ROOT_URLCONF='admin_views.urls')
+class SeleniumTests(AdminSeleniumTestCase):
+    available_apps = ['admin_views'] + AdminSeleniumTestCase.available_apps
+
+    def setUp(self):
+        self.superuser = get_user_model().objects.create_superuser(
```

Comment:
No need to use `get_user_model()` (don't think any of the other tests do that?), I think.